types/connector.d.ts

```diff
@@ -1,16 +1,14 @@
-import { URL } from 'url'
-import { TLSSocket, TlsOptions } from 'tls'
-import { Socket } from 'net'
+import {TLSSocket, ConnectionOptions} from 'tls'
+import {NetConnectOpts, Socket} from 'net'
 
 export = buildConnector
 declare function buildConnector (options?: buildConnector.BuildOptions): typeof buildConnector.connector
 
 declare namespace buildConnector {
-  export interface BuildOptions extends TlsOptions {
+  export type BuildOptions = (ConnectionOptions | NetConnectOpts) & {
```

Comment:
```suggestion
  export interface BuildOptions extends ConnectionOptions, NetConnectOpts {
```

see: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections